### ISSUE TITLE
Using `group` + `color` produces incorrect plot

### ISSUE THREAD

**Issue by alyst** [ropensci/plotly#418]
`plotly` incorrectly draws lines `scatter` (and `scatter3d`) plots.

Consider the following minimal example:
```R
df = data.frame(x=c(1,2,3,4,5,6),
           y=c(1,1,2,2,3,3),
           t=c("A","A","B","B","A","A"))

plot_ly(df, type="scatter", x=x, y=y, color=t, group=y, mode="lines")
````

The output is:
![image](https://cloud.githubusercontent.com/assets/348591/12484986/08bd8212-c05d-11e5-9d3c-61b560a47bef.png)

In the correct plots only the horizontal segments colored according to `t` column are expected; also no `1`,`2`,`3` entries in the legend, `A` and `B` sorted either alphabetically or by their appearance in the data frame.

**Comment by cpsievert** [comment]
I have a note of this here as well #381

I hope to get to this soon...thanks for the minimal example!

**Comment by alyst** [comment]
@cpsievert Yes, it's the same as #381 + an additional issue of drawing "parasitic line" `(2,1) - (5,3)`.

**Comment by cpsievert** [comment]
Closed via d2bf2a1eb56d824a091e2cc9afea09a5968dd050. This is now the official way to do this:

```r
df = data.frame(x=c(1,2,3,4,5,6),
           y=c(1,1,2,2,3,3),
           t=c("A","A","B","B","A","A"))

df %>% group_by(y) %>% plot_ly(x = ~x, y = ~y, color = ~t) %>% add_lines()
```
